test/viewport.js

```diff
@@ -276,6 +278,299 @@
     });
 
+    asyncTest('resetContentSize', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+
+            for(var i = 0; i < testRects.length; i++){
+                var rect = testRects[i].times(viewport.getContainerSize());
+                viewport.resetContentSize(rect);
+                viewport.update();
+                propEqual(
+                    viewport.contentSize,
+                    rect,
+                    "Reset content size correctly."
+                );
+            }
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(DZI_PATH);
+    });
+
+    asyncTest('goHome', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+
+            // zoom/pan somewhere
+            viewport.zoomTo(ZOOM_FACTOR, true);
+            viewport.update();
+
+            viewport.goHome(true);
+            viewport.update();
+            propEqual(
+                viewport.getBounds(),
+                viewport.getHomeBounds(),
+                 "Went home."
+            );
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(DZI_PATH);
+    });
+
+    asyncTest('ensureVisible', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+
+            // zoom/pan so that the image is out of view
+            viewport.zoomTo(ZOOM_FACTOR * -50, true);
+            viewport.panBy(new OpenSeadragon.Point(5000, 5000), null, true);
+            viewport.update();
+
+            viewport.ensureVisible(true);
+            viewport.update();
+            var bounds = viewport.getBounds();
+            ok(bounds.getSize().x > 1 && bounds.getSize().y > 1, "Moved viewport so that image is visible.");
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(DZI_PATH);
+    });
+
+    asyncTest('fitBounds', function(){
+        var openHandler = function(event) {
+            viewer.removeHandler('open', openHandler);
+            var viewport = viewer.viewport;
+
+            for(var i = 0; i < testRects.length; i++){
+                var rect = testRects[i].times(viewport.getContainerSize());
+                viewport.fitBounds(rect, true);
+                viewport.update();
+                propEqual(
+                    viewport.getBounds(),
+                    rect,
+                    "Fit bounds correctly."
+                );
+            }
+            start();
+        };
+        viewer.addHandler('open', openHandler);
+        viewer.open(DZI_PATH);
+    });
+
+    var testRectsOutsideImage = [
+        new OpenSeadragon.Rect(0, 0, 2000, 100),
```

Comment:
These rects should be in the viewport coordinate system, where 0 is the left side of the image and 1 is the right side. The values you're giving here are way out of range. 

Also, if you give values that are so much bigger than the image, its only option is to fit the whole image (which is why you're always getting the same value). To test the constraints, you'll want to try things like `0.75, 0.25, 0.5, 0.5`, where the rect you're asking for is smaller than the image but hanging a little off of the side of it.